gulpfile.js

```diff
@@ -38,7 +38,7 @@ gulp.task('ghpages', 'Deploy documentation to gh-pages', ['nojekyll'], function
 
 gulp.task("docs", 'Compile documentation from src code', function () {
   return gulp
-    .src(["src/**/*.ts"])
+    .src(["typings/globals/es6-promise/index.d.ts", "src/**/*.ts"])
```

Comment:
Big reason for updating to `IFilter` was to support typedoc.  Typedoc doesn't support the (A|B) syntax.  If we include all the typings then it will generate docs for dev dependencies like jasmine, etc.